### PR TITLE
feat: LlamaIndex Workflows on Cloud Run via Llama Deploy

### PR DESCRIPTION
**Pull request by noabenefraim**
# Description

Thank you for opening a Pull Request!
Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [ ] Follow the [`CONTRIBUTING` Guide](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/CONTRIBUTING.md).
- [ ] You are listed as the author in your notebook or README file.
  - [ ] Your account is listed in [`CODEOWNERS`](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/.github/CODEOWNERS) for the file(s).
- [ ] Make your Pull Request title in the <https://www.conventionalcommits.org/> specification.
- [ ] Ensure the tests and linter pass (Run `nox -s format` from the repository root to format).
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: gemini/sample-apps/llamadeploy-on-cloud-run/data/gatsby.txt
Comment (holtskinner) [inline gemini/sample-apps/llamadeploy-on-cloud-run/data/gatsby.txt:1]: Please remove this file from the PR. Large data files like this should be uploaded to Google Cloud Storage. I added this to our public GCS bucket here:

`gs://github-repo/generative-ai/sample-apps/llamadeploy-on-cloud-run/gatsby.txt`

Either have the user download it with gsutil or send the gcs path directly to gemini if possible

File: .github/actions/spelling/excludes.txt [hunk 1]
```diff
@@ -110,3 +110,4 @@ ignore$
 ^\Qsearch/bulk-question-answering/bulk_question_answering_output.tsv\E$
 ^\Q.github/workflows/issue_assigner/assign_issue.py\E$
 ^\Qnoxfile.py\E$
+^\Qgemini/sample-apps/llamadeploy-on-cloud-run/data/gatsby.txt\E$
```
Comment (holtskinner) [inline .github/actions/spelling/excludes.txt:113]:
```suggestion
```

**Review by holtskinner — CHANGES_REQUESTED** [review]

File: gemini/sample-apps/llamadeploy-on-cloud-run/data/gatsby.txt
Comment (noabenefraim) [reply gemini/sample-apps/llamadeploy-on-cloud-run/data/gatsby.txt:1]: Removed! Added download instructions to the README.md.

File: .github/actions/spelling/excludes.txt (hunk 1 above)
Comment (noabenefraim) [reply .github/actions/spelling/excludes.txt]: Removed.

File: gemini/sample-apps/llamadeploy-on-cloud-run/workflow.py [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+        new_nodes: List[NodeWithScore] = []
+
+        text_splitter = SentenceSplitter(
+            chunk_size=DEFAULT_CITATION_CHUNK_SIZE,
+            chunk_overlap=DEFAULT_CITATION_CHUNK_OVERLAP,
+        )
+
+        for node in nodes:
+
+            print(node)
```
[the remark this answers is not in the document]
Comment (noabenefraim) [reply gemini/sample-apps/llamadeploy-on-cloud-run/workflow.py:391]: Good for the logs, to see what data was retrieved.

File: gemini/sample-apps/llamadeploy-on-cloud-run/workflow.py [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+            )
+
+            for text_chunk in text_chunks:
+                text = f"Source {len(new_nodes)+1}:\n{text_chunk}\n"
+
+                new_node = NodeWithScore(
+                    node=TextNode.model_validate(node.node), score=node.score
+                )
+                new_node.node.text = text
+                new_nodes.append(new_node)
```
[the remark this answers is not in the document]
Comment (noabenefraim) [reply gemini/sample-apps/llamadeploy-on-cloud-run/workflow.py:404]: I don't think so. ```model_validate``` only takes in an object, but I can't further assign variable to the object being validated.

File: gemini/sample-apps/llamadeploy-on-cloud-run/workflow.py [hunk 4]
```diff
@@ -0,0 +1,450 @@
...
+        # should_stop = False
+        cur_steps = 0
+
+        # # use response
+        # final_response_metadata: Dict[str, Any] = {"sub_qa": []}
+
+        # text_chunks: list[str] = []
+        # source_nodes: list[NodeWithScore] = []
+
+        # stop_fn = self.default_stop_fn
```
[the remark this answers is not in the document]
Comment (noabenefraim) [reply gemini/sample-apps/llamadeploy-on-cloud-run/workflow.py:302]: Done.

File: gemini/sample-apps/llamadeploy-on-cloud-run/workflow.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (noabenefraim) [reply gemini/sample-apps/llamadeploy-on-cloud-run/workflow.py:380]: Resolving for now, feel free to unresolve @holtskinner

**Review by holtskinner — APPROVED** [review]

**Pull request merged**